smartconfig/file.py

```diff
@@ -77,6 +77,19 @@ def load(path: Union[str, bytes, PathLike]) -> None:
     a `ConfigEntry`. There could be nodes that are simply not used at all. It's even valid for the root to not be a
     mapping node. Such file would effectively configure nothing, but loading it is still supported.
 
+    A special !REF construct can be used to point to another attribute. For instance,
+
+    module_1:
+      class:
+        attribute_1: value
+
+    module_2:
+      class:
+        reference: !REF module_1.class.attribute_1
+
+    module_2.class.reference will have the same value as module_1.class.attribute_1 even if it is later overwrote by
+    a configuration file.
```

Comment:
It's not clear that the referenced attribute need not exist in the config (in which case it'd use the default value of the class). Also, in this case, it should say _overwritten_ not _overwrote_ (though I find the entire "even if..." part to be confusing anyway).

We should try to come up with a way to explain it in terms of `ConfigEntry` attributes, since that is well documented above. If we just say it references the class's attribute, then they can refer to the above documentation for how that referenced attribute's value is determined.

Reply:
Resolved by dda2586